app/models/application_record.rb

```diff
@@ -1,3 +1,16 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+  default_scope -> { order(date: :desc) }
+
+  # def self.past
+  #   Event.all.where("date < ?", Time.now)
+  # end
+
+  scope :past, -> { where('date < ?', Time.now) }
```

Comment:
- [ ] Why are you putting scopes for past and upcoming events in application record model? It should be in `event` model. This will still work, because all models are inheriting from applictionrecord, but this breaks encapsulation.